connectome_interpreter/utils.py

```diff
@@ -1087,3 +1087,164 @@ def path_for_ngl(path):
         df = df.drop_duplicates()
         dfs.append(df)
     return pd.concat(dfs)
+
+
+def make_grid_inputs(v1, v2, num_layers: int, grid_size: int = 10, timepoints: int | list = 0, device=None) -> Tuple[torch.Tensor, List[Tuple[float, float]]]:
+    """
+    Make a batch of input using combinations of v1 and v2 at different strengths (0 to 1). 
+
+    Args:
+        v1 (np.ndarray): The first input vector.
+        v2 (np.ndarray): The second input vector. Its length should match that of `v1`.
+        num_layers (int): The number of layers in the model.
+        grid_size (int, optional): The number of points in the grid. Defaults to 10.
+        timepoints (int|list, optional): The timepoints at which combinations of v1 and v2 are used. Defaults to 0 (the first timepoint).
+        device (str, optional): The device to create the inputs on. If None, if GPU is available, the inputs are created on the GPU. Otherwise CPU.
+
+    Returns:
+        torch.Tensor: A tensor of shape (grid_size**2, len(v1), num_layers).
+        list: A list of grid coordinates.
+    """
+
+    # first check if length of v1 and v2 are the same
+    if len(v1) != len(v2):
```

Comment:
do you need them to be 1-dimensional too?

Reply:
Good point, modified.